### ISSUE TITLE
Enable metadata saving via TensorFlow.js frontend API

### ISSUE THREAD

**Issue by jasonmayes** [tensorflow/tfjs#6414]
Currently metadata is supported for conversion via the CLI converter on server side from models from Python however if you train a model in the browser there is no way to save custom key value pairs that you may need in future pre/post processing.

For example if you perform normalization on your training data you need to know the min/max values for a given attribute in order to normalize future data after the model is trained too at inference for any new prediction data that comes in.

Currently a user has to save this data in a different pipeline to the TensorFlow.js one which is inconvenient and does not make sense. It would be far more optimal to expose a simple interface to the end user to store their metadata as part of the model.json file and then able to read these values on model load such that they can be retrieved and used as needed for any custom pre/post processing when using the model.

All users training models in the browser either fully or via transfer learning could benefit from this feature.

**Comment by halflings** [comment]
For the specific use case of pre-processing, it would be great to have ops like the ones in tf transform:
https://www.tensorflow.org/tfx/transform/get_started

Normalizing a feature would just require calling something like `tft. scale_to_z_score(..)` when building the model, and the rest is done more-or-less transparently (no need to manually serialize/deserialize metadata).

Here are some of the most common pre-processing steps I've used (either for MLConsole.com using custom JS code, or at work with TFX):
* Scaling numeric variables (metadata = mean, variance ; or min,max)
* Mapping categorical variables to an embedding ID (metadata = vocab)
* Tokenizing strings. (little to no metadata, but good to have as part of the graph ; usually combined with the mapping to embedding ID)

**Comment by gaikwadrahul8** [comment]
Hi, @jasonmayes 

Apologize for the delayed response and It seems like we have implemented this feature in `Tensorflow.js` If I'm not wrong so could you please close this issue if we have implemented this feature in `Tensorflow.js `? 

If have I missed something here or we haven't implemented this feature in `Tensorflow.js`, Please let me know ? Thank you!

**Comment by jasonmayes** [comment]
It was? Can you point me to the docs or PR or such? I did not receive any update on this being implemented yet.

**Comment by gaikwadrahul8** [comment]
Hi, @jasonmayes 

I am truly sorry for for my previous misunderstanding of this feature request. After further investigation, I believe that there is currently no Tensorflow.js frontend api to enable metadata saving. This would allow end users to store their metadata as part of the` model.json` file and then read these values on model load. This would be useful for retrieving and using the metadata as needed for any custom pre- or post-processing steps when using the model. Thank you!

**Comment by jasonmayes** [comment]
@mattsoulanille Did you ever look into this by chance? I can not quite remember if we discussed this in one of our 1-1s a while back.

**Comment by nileshau-afk** [comment]
Hi @jasonmayes  ,

We are closing this issue due to inactivity. If this is still relevant and you would like to work on it, please feel free to reopen the issue or submit a pull request referencing it.
Thank you for your understanding.
